2016/proposals/vora-meet.md

```diff
@@ -0,0 +1,161 @@
+ * For every *respondent_id* in *respondent_ids*, we iterate over all *question_id* and for a given question, the *count* is calculated against each *answer_id*.
+ * The calculated result is saved as a JSON object and stored in the database.
+
+The structure of the model used:
+```python
+
+class Workshop(models.Model):
+	all_survey_ids = [112, 123, 235, 358, 5813]		# random values chosen for demonstration purpose
+	SURVEYS = ( 									# linking type of survey to the survey_ids 
+			(all_survey_ids[0], 'PreWorkshop Survey'),
+			(all_survey_ids[1], 'PostWorkshop Survey'),
+			(all_survey_ids[2], 'Organizer Survey 1'),
+			(all_survey_ids[3], 'Organizer Survey 2'),
+			(all_survey_ids[4], 'Longterm Survey'),
+		)
+	id = models.CharField(max_length=200)
+	last_access = models.DateField()
+	survey_id = models.IntegerField(choices = SURVEYS)
+	host = models.CharField(max_length=500) 
+	respondent_ids = models.CharField(max_length=10000)
+	json_string = models.TextField()
+	url = models.CharField(max_length=500)
+	date_conducted = models.DateField()
+```	  
+
+The final calculated list of questions for a particular survey data will store JSON objects as: 
+```python
+[	
+	{
+		'id': 		question_id,
+		'text': 	question_text,
+		'type':		question-type # e.g. 'single-choice-radio' or 'matrix-menu'
+		'answer':	[
+						{
+							'id': 		answer_id
+							'text':		answer_text
+							'count':	answer_count['count']	 	# answer_count shall be user-defined dictionary as described above
+						},
+					]
+	},
+]
+```
+
+###### UML Sequence Diagram
+![Sequence Diagram](https://lh3.googleusercontent.com/ly5wZV2HjIlDMSuPA6evW-a-mdphQ2X4ZAdJWyfCxhNU3GbMPNauPSMf2gEeBYQ-JqA6h8BZ02-dYad-hnwubrO5hTir4Y-4aEd0FNHIPDPKorhbcY6hB_azuiVnwcgMCwR1oYN9XmSwFw1YRoxYMQ3Bz-p_yzQik4m55kimcMyiSjhoW-NYFtjBrksbvg4wIkFQ7YfBIUNbW3ezaOILfRqMZsCU6YWOR-rlMuozt1VmUyCM8hnvYrMlgjqX8xdKEitSBYWGp0quY0RccgB03IiIP3w4uvy5h8nc7dp49D9Whb39msXY84e_TC2jaYKeGJHfDJujpGf0mtS7-gZts2f3J2UXnaI7_4ney6qXqXwpbRYME4zLWeCL4cX81KbXJmk_IqDQSCBePQdO-0ZwcKIh0CscmQf4gaQIpF1Mw-yehHuKhP99VJ0G282N9xnCAytkl-mC93AZxB_NOWAgkJEB7cny_dBbulksz4C565mUzXMduhEEh8x6ax503EamMOtt-PGGdprPNbBOKn9KfxAmmCs4RK9ct0u3yq5B9vHNCATP0f16NYL8WZVGwN8=w777-h677-no)
+
+#### **Database**: PostgreSQL
+With the release of v9.4, extensive support for JSON has been added to PostgreSQl, making it the perfect solution for our idea. PostgreSQL can easily handle large amounts of data and thus, speed performance and fetching of newer data wouldn't be an issue.  
+Further, as per the discussions with the mentors, I have been made aware that SQLite is currently being employed in AMY. Thus, we can currently store the JSON as a string in PostgreSQL, to maintain cross-database compatiblity.  
+To enable support for simple queries, we will save certain data (e.g. *date of workshop*) in seperate fields in the models.
```

Comment:
Okay, my guts tell me the most crucial part of the proposal lies in the good database schema.

Please:

1. expand your SQLite / cross-db compatible design
2. expand your PostgreSQL-only design

You can argue which one you prefer and why.

Thanks!